pkgs/servers/nextcloud/packages/generate.sh

```diff
@@ -8,7 +8,55 @@ set -x
 
 export NEXTCLOUD_VERSIONS=$(nix-instantiate --eval -E 'import ./nc-versions.nix {}' -A e)
 
-APPS=`cat nextcloud-apps.json | jq -r 'keys|.[]' | sed -z 's/\n/,/g;s/,$/\n/'`
+APPS=$(jq -r 'keys|.[]' nextcloud-apps.json | sed -z 's/\n/,/g;s/,$/\n/')
 
-nc4nix -apps $APPS
-rm *.log
+for v in ${NEXTCLOUD_VERSIONS//,/ }; do
+    # Get major version and back up previous major version apps file
+    v=$(sed -e 's/^"//' -e 's/"$//' <<<"$v")
+    MAJOR=${v%.*.*}
+    MAJOR_FILE="$MAJOR".json
+    mv "$MAJOR_FILE" "$MAJOR_FILE".bak
+
+    # Download current apps file from Nextcloud's official servers
+    APPS_PER_VERSION=${v}.json
+    curl "https://apps.nextcloud.com/api/v1/platform/${v}/apps.json" -o "$APPS_PER_VERSION"
+
+    # Add a starting bracket to the apps file for this version
+    echo '{' >"$MAJOR_FILE".tmp
+    for a in ${APPS//,/ }; do
+        echo "Fetching $a"
+        # Ensure the app exists in the file
+        if [ "$(jq -r ".[] | select(.id == \"${a}\")" "$APPS_PER_VERSION")" != "" ]; then
+            # Get all of our variables
+            VERSION=$(jq -r ".[] | select(.id == \"${a}\") | .releases[0].version" "$APPS_PER_VERSION")
+            URL=$(jq -r ".[] | select(.id == \"${a}\") | .releases[0].download" "$APPS_PER_VERSION")
+            HASH=$(nix store prefetch-file --json --hash-type sha256 --unpack "$URL" | jq -r .hash)
```

Comment:
This raises
```
[…]
++ jq -r .hash                                                                                                                                                                                    
++ nix store prefetch-file --json --hash-type sha256 --unpack https://github.com/nextcloud/bookmarks/releases/download/v14.2.2/bookmarks-14.2.2.tar.gz                                            
error: unrecognised flag '--unpack' 
```
when I try to run the generation script. How does this work? My nix version is `nix (Nix) 2.18.4
`.

Reply:
Ah, --unpack is nix 2.20.0/lix-exclusive. I'll send a followup that uses nix-prefetch-url instead, which has a backwards-compatible --unpack option that nix 2.18 can use.

Reply:
FWIW I just fixed this up locally by requiring `nixVersions.latest` in the `generate.sh` script.

Reply:
I think we want to pin Nix to an explicit version here. Technically, you can still use Nix 2.3 to instantiate nixpkgs and requiring newer features is fine here, but it has to be made explicit.